### ISSUE TITLE
Directory structure highlighting

### ISSUE THREAD

**Issue by kbjr** [PrismJS/prism#193]
I created a new syntax file for highlighting plain-text directory structure diagrams, like this:

```
|-- app
|   |-- controllers
|   |-- models
|   `-- views
```

It's extremely simple, just a single small regex to highlight commonly used symbols for this purpose:

```javascript
Prism.languages.folders = {
    // Just highlight symbols used to denote folder structure
    keyword: /^([-|+`\s]+)/gm
};
```

Is this something you think others might want? I can give you a pull request very easily if you think so, I just wasn't sure if anyone other than me cares :p

**Comment by loganfranken** [comment]
This is awesome @kbjr and @Golmote!

It doesn't look like there are plans to bring this into the core Prism, right? If not, we could probably close this issue,

**Comment by Golmote** [comment]
You're right.

**Comment by muescha** [comment]
is there any reason why it is not in core? i think it's a very common need to show file structures in code blocks

**Comment by LeaVerou** [comment]
Hi @muescha,
We generally try to keep the core very small. Is there a reason you need this in the core? With the bundler on the website, there isn't really much difference.

**Comment by Golmote** [comment]
@LeaVerou He meant why is it an external project. But we now have included this feature as a plugin in #2265! 🎉
